creator-node/src/snapbackSM/computeSyncModeForUserAndReplica.js

```diff
@@ -0,0 +1,115 @@
+const DBManager = require('../dbManager.js')
+const { asyncRetry } = require('../utils.js')
+
+/**
+ * Sync mode for a (primary, secondary) pair for a user
+ */
+const SyncMode = Object.freeze({
+  None: 'NONE',
+  SecondaryShouldSync: 'SECONDARY_SHOULD_SYNC',
+  PrimaryShouldSync: 'PRIMARY_SHOULD_SYNC'
+})
+
+const FetchFilesHashNumRetries = 3
+
+/**
+ * Given user state info, determines required sync mode for user and replica. This fn is called for each (primary, secondary) pair
+ * @notice Is used when both replicas are running version >= 0.3.51
+ * @param {Object} param
+ * @param {string} param.wallet user wallet
+ * @param {number} param.primaryClock clock value on user's primary
+ * @param {number} param.secondaryClock clock value on user's secondary
+ * @param {string} param.primaryFilesHash filesHash on user's primary
+ * @param {string} param.secondaryFilesHash filesHash on user's secondary
+ * @returns {SyncMode} syncMode one of None, SecondaryShouldSync, PrimaryShouldSync
+ */
+async function computeSyncModeForUserAndReplica({
+  wallet,
+  primaryClock,
+  secondaryClock,
+  primaryFilesHash,
+  secondaryFilesHash,
+  logger
+}) {
+  if (
+    // clock value must be number
+    !(primaryClock && primaryClock !== 0) ||
+    !(secondaryClock && secondaryClock !== 0) ||
+    // `null` is a valid filesHash value
+    primaryFilesHash === undefined ||
+    secondaryFilesHash === undefined
+  ) {
+    throw new Error(
+      '[computeSyncModeForUserAndReplica] Error: Missing or invalid params'
+    )
+  }
+
+  let syncMode = SyncMode.None
+
+  if (primaryClock === secondaryClock) {
+    if (primaryFilesHash === secondaryFilesHash) {
+      syncMode = SyncMode.None
+    } /* primaryFilesHash !== secondaryFilesHash */ else {
+      syncMode = SyncMode.PrimaryShouldSync
```

Comment:
just to be clear, this case is if there's a divergence in hash because primary is missing data, so primary should sync and should obtain that data and in the future ask secondary to resync?

Reply:
if that's the case could you comment that here

Reply:
+1, can we add comments to the SyncMode const?

Reply:
yep @dmanjunath thats correct, and ye great point, added comment